yarn-project/simulator/src/public/public_tx_simulator.ts

```diff
@@ -385,7 +386,15 @@ export class PublicTxSimulator {
    */
   public async insertNonRevertiblesFromPrivate(context: PublicTxContext) {
     const stateManager = context.state.getActiveStateManager();
-    await stateManager.writeSiloedNullifiersFromPrivate(context.nonRevertibleAccumulatedDataFromPrivate.nullifiers);
+    try {
+      await stateManager.writeSiloedNullifiersFromPrivate(context.nonRevertibleAccumulatedDataFromPrivate.nullifiers);
+    } catch (e) {
+      if (e instanceof NullifierCollisionError) {
```

Comment:
What I don't like is this trims the stack trace of the other error, maybe we could include e.stack

Reply:
Hmm good point.... I was just trying to enrich the message a bit with more context. But that's context that you can kind of figure out by looking at the original stack trace.

Ultimately, we need this try catch anyway. I think we ultimately _don't_ want the transaction to be dropped due to a duplicate nullifier coming from private. Instead I think we want to have the transaction be "reverted" in the proper phase with its nullifiers dropped, but it can still be included in the block and the sender pays gas.